starknet-rpc-test/get_class.rs

```diff
@@ -47,58 +61,158 @@ async fn fail_non_existing_class_hash(madara: &ThreadSafeMadaraClient) -> Result
-        FieldElement::from_hex_be(TEST_CONTRACT_CLASS_HASH).expect("Invalid Contract Class Hash");
-
-    let test_contract_class_bytes = include_bytes!("../cairo-contracts/build/test.json");
-    let test_contract_class: LegacyContractClass = serde_json::from_slice(test_contract_class_bytes).unwrap();
+    let unknown_contract_address = FieldElement::from_hex_be("0x4269DEADBEEF").expect("Invalid Contract classh hash");
 
     assert_matches!(
-        rpc
-        .get_class(
-            BlockId::Number(0),
-            test_contract_class_hash,
-        ).await?,
-        ContractClass::Legacy(c) => {
-            // decompress program
-            let mut gz = GzDecoder::new(&c.program[..]);
-            let mut decompressed_bytes = Vec::new();
-            gz.read_to_end(&mut decompressed_bytes).unwrap();
-            let legacy_program_wrapper: LegacyProgramWrapper = serde_json::from_slice(decompressed_bytes.as_slice())?;
-            let program = legacy_program_wrapper.legacy_program;
-            assert_eq!(
-                program.data.len(),
-                test_contract_class.program.data.len(),
-            );
-        }
+        rpc.get_class_at(BlockId::Number(0), unknown_contract_address,).await,
+        Err(StarknetProviderError(StarknetError::ContractNotFound))
     );
 
     Ok(())
 }
 
-#[ignore = "conversion between contract class types is incomplete"]
 #[rstest]
 #[tokio::test]
-async fn work_ok_retrieving_class_for_contract_version_1(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+async fn work_ok_retrieving_class_for_contract_version_0(madara: &ThreadSafeMadaraClient) {
+    // Test that the contract class received from the rpc call get_class and get_class_at is the same as
+    // the one that was used to declare
+    fn assert_eq_contract_class(received: ContractClass, mut expected: LegacyContractClass) {
+        assert_matches!(
+            received,
+            ContractClass::Legacy(c) => {
+
+                assert_eq!(
+                    serde_json::to_value(expected.abi).unwrap(),
+                    serde_json::to_value(c.abi).unwrap(),
+                );
+                assert_eq!(
+                    serde_json::to_value(expected.entry_points_by_type).unwrap(),
+                    serde_json::to_value(c.entry_points_by_type).unwrap(),
+                );
+
+                // decompress program
+                let mut gz = GzDecoder::new(&c.program[..]);
+                let mut decompressed_bytes = Vec::new();
+                gz.read_to_end(&mut decompressed_bytes).unwrap();
+                let mut program: LegacyProgram = serde_json::from_slice(decompressed_bytes.as_slice()).unwrap();
+
+                // Because of some obscure pathfinder bug, starknet-rs deserialize debug_info as an empty struct rather than none
+                // It has been fixed, so when https://github.com/xJonathanLEI/starknet-rs/pull/599 is megerd, this will become useless
```

Comment:
merged no?

Reply:
yup, it is now. But we will have to wait for a new release of starknet-rs